src/hotspot/share/gc/shared/ageTable.hpp

```diff
@@ -52,17 +53,18 @@ class AgeTable {
 
   // clear table
   void clear();
+  // check whether it's clear
+  bool is_clear() PRODUCT_RETURN0;
```

Comment:
Thanks for taking a look, Albert.

It was indeed intended just for verification/assertion checks.

The use case(s) can be found here in the GenShen repo:

https://github.com/openjdk/shenandoah/blob/5c6f4171efd250ab47695059814a78d7b672a4bd/src/hotspot/share/gc/shenandoah/shenandoahAgeCensus.cpp#L190

and other spots in the same file. They are all for assertion checks. 

I expect that this will be used only for assertions, whenever it is, hence non-product.

Reply:
It's not obvious to me that `bool clear = _global_age_table[i]->is_clear();` is for assert-only. Looking at `ShenandoahAgeCensus::is_clear_global`, the whole methods looks like it's real/product logic.

> I expect that this will be used only for assertions, whenever it is, hence non-product.

Then, can its definition be guarded like following? My concern is that this API returns different value depending on whether it is a product build.

```
#ifdef ASSERT // or ifndef product
bool is_clear();
#endif
```

(Or, leaving it as an ordinary API seems OK as well, IMO.)

Reply:
Fair point. I'll fix this.

Reply:
Made it #ifndef PRODUCT.